docs/tutorials/snippet/admin-bar-best-practices.md

```diff
@@ -0,0 +1,438 @@
+
+<AposCodeBlock>
+
+```javascript
+export default {
+  options: {
+    // Prevent this content type from appearing in the quick create menu
+    quickCreate: false
+  }
+};
+```
+  <template v-slot:caption>
+    modules/article/index.js
+  </template>
+</AposCodeBlock>
+
+**When to remove items:**
+- Rarely created content types
+- Specialized content for specific roles only (even if only selected roles can create a specific piece-type, the menu shouldn't be over-cluttered)
+- When you have many piece types and need to reduce options
+- Content types requiring careful setup that should use the full content manager
+
+## Customizing the Context Utility Menu
+
+Add custom utilities to the right side of the admin bar from the `init` method of any module:
+
+```javascript
+export default {
+  init(self) {
+    // Add item to the context utility menu (right side of admin bar)
+    self.apos.adminBar.add(
+      'custom-module:help', // Unique id
+      'myproject:getSupport', // Label
+      false,                 // Permissions (false = available to all)
+      {
+        contextUtility: true,  // This makes it appear in the right side
+        icon: 'question-circle', // Icon to display (required)
+      }
+    );
+  }
+};
+```
+> [!NOTE]
+> Custom actions or link navigation need to be added through browser-side scripting as shown in the [Adding Custom Action Buttons](#adding-custom-action-buttons) section.
+
+**Best uses for context utilities:**
+- Universal features needed from any context
+- Support tools that should never be more than one click away
+- Project-specific tools that don't fit the standard content workflow
+
+## Customizing the User Dropdown Menu
+
+### Adding Items to the Menu
+
+You can add custom items to the user dropdown menu (where "Log Out" appears) using the `adminBar.add()` method with the `user: true` option:
+
+<AposCodeBlock>
+
+```javascript
+export default {
+  init(self) {
+    // Add a custom item to the user menu
+    self.apos.adminBar.add(
+      'custom-module:user-dashboard',             // Unique identifier
+      'myproject:userDashboard',    // Translation string
+      false,                        // Permissions (false = available to all)
+      {
+        user: true,                 // Places this in the user menu
+      }
+    );
+  }
+};
+```
+  <template v-slot:caption>
+    modules/my-custom-module/index.js
+  </template>
+</AposCodeBlock>
+
+> [!NOTE]
+> Items added to the user menu cannot be grouped further. When a user clicks this menu item, an event will be emitted on `apos.bus` with the name you provided ('custom-module:user-dashboard' in this example). You'll need to add a corresponding event handler in your frontend code to respond when the item is clicked as shown in the prior examples.
+
+### Modifying the User Preferences Menu
+
+Give content managers control over their experience through customized preferences:
```

Comment:
This feature can be used to allow users to edit any field of their own user piece, including useful ones out of the box like `password` and `title` that might be good starting points because you don't have to gesture at "... and now build custom functionality so that does something." Although it's good to do that too.

(There's a short list of forbidden fields to avoid scenarios where people can impersonate or become an admin.)